### PR TITLE
Add option for using p2p ops comm. for actv

### PR DESCRIPTION
**Pull request by gajagajago**
Current activation sdrv relies on batch_p2p (when `overlap_p2p` option is given).
This makes wait on its handle to also wait for isend, which is not necessary.
This PR removes such behavior, only blocking on the wait() for the irecv handle of activation communication. 

This commit is rebased on top of PR #48. Please merge 48 first and then this.

### CODE REVIEW

**Comment by ttochi** [comment]
In original code, if I give `--overlap-p2p-comm` option, doesn't `batched_p2p_comm == False`?
(even if we don't create `--spiral-actv-p2p` option)

**Comment by gajagajago** [comment]
To allow overlap_p2p && batch_p2p_comm, we add new option, `spiral-actv-p2p`. Also, we add send request omitting option.

File: megatron/training.py [hunk 1]
```diff
@@ -840,7 +840,7 @@ def train_step(forward_step_func, data_iterator,
         grad_scaler=getattr(optimizer, "scale_loss", None),
         sequence_parallel=args.sequence_parallel,
         overlap_p2p_comm=args.overlap_p2p_comm,
-        batch_p2p_comm=not args.overlap_p2p_comm,
+        batch_p2p_comm=not args.overlap_p2p_comm and not args.spiral_actv_p2p,
```
Comment (ttochi) [inline megatron/training.py]: overlap_p2p_comm = True
spiral_actv_p2p = False

If you intended `batch_p2p_comm = True` for this case, should it be `or`?

Comment (gajagajago) [reply megatron/training.py]: oops ~

Comment (gajagajago) [inline megatron/training.py]:
```suggestion
        batch_p2p_comm=not args.overlap_p2p_comm or not args.spiral_actv_p2p,
```

**Review by ttochi — APPROVED** [review]
LGTM

**Pull request merged**